identity/src/main/java/life/qbic/identity/domain/model/token/PersonalAccessTokenEncoder.java

```diff
@@ -0,0 +1,172 @@
+package life.qbic.identity.domain.model.token;
+
+import static java.util.Objects.requireNonNull;
+import static life.qbic.logging.service.LoggerFactory.logger;
+
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+import life.qbic.logging.api.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Encodes personal access tokens
+ */
+@Service
+public class PersonalAccessTokenEncoder implements TokenEncoder {
+
+  private static final Logger log = logger(PersonalAccessTokenEncoder.class);
+
+  private final String salt;
+  private final int iterationCount;
+
+  private static final int EXPECTED_MIN_ITERATION_COUNT = 10_000;
+
+  private static final int ITERATION_COUNT_INDEX = 0; // the index of the iteration count in the encoded token
+  private static final int SALT_INDEX = 1; // the index of the salt content in the encoded token
+  private static final int HASH_INDEX = 2; // the index of the hash content in the encoded token
+
+  public PersonalAccessTokenEncoder(
+      @Value("${qbic.access-token.salt}") String salt,
+      @Value("${qbic.access-token.iteration-count}") int iterationCount) {
+    this.salt = requireNonNull(salt, "salt must not be null");
+    if (this.salt.isBlank()) {
+      throw new IllegalArgumentException("salt must not be blank");
+    }
+    if (iterationCount <= 0) {
+      throw new IllegalArgumentException("iterationCount must be greater than 0");
+    }
+    if (iterationCount <= EXPECTED_MIN_ITERATION_COUNT) {
+      log.warn(
+          "Low iteration count for token encryption detected n=%d. Expected minimal iteration count is n=%d".formatted(
+              iterationCount, EXPECTED_MIN_ITERATION_COUNT));
+    }
+    this.iterationCount = iterationCount;
+  }
+
+  private record EncryptionSettings(String cipher, int keyBitSize) {
+
+  }
+
+  private static final EncryptionSettings ENCRYPTION_SETTINGS = new EncryptionSettings("AES", 256);
+
+
+  @Override
+  public String encode(char[] token) {
+    String saltyString = requireNonNull(this.salt, "this.salt must not be null");
+    if (saltyString.isBlank()) {
```

Comment:
This is not enough ("All or a portion of the salt shall be generated using an approved Random Bit Generator
(e.g., see [5]). The length of the randomly-generated portion of the salt shall be at least
128 bits", NIST recommendation: https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-132.pdf),

Please ensure that the byte array of the encoded value is of size 16 or longer.